In [6]:
import numpy as np

marks = np.array([78, 85, 92, 67, 88, 95, 73, 81, 90, 76])

print("Student Marks:", marks)

Student Marks: [78 85 92 67 88 95 73 81 90 76]


In [7]:
mean_marks = np.mean(marks)
median_marks = np.median(marks)

print("Mean Marks:", mean_marks)
print("Median Marks:", median_marks)

Mean Marks: 82.5
Median Marks: 83.0


In [8]:
maximum_marks = np.max(marks)
minimum_marks = np.min(marks)

print("Maximum Marks:", maximum_marks)
print("Minimum Marks:", minimum_marks)

Maximum Marks: 95
Minimum Marks: 67


In [9]:
mean_marks = np.mean(marks)
median_marks = np.median(marks)

print("Mean Marks:", mean_marks)
print("Median Marks:", median_marks)

Mean Marks: 82.5
Median Marks: 83.0


In [10]:
maximum_marks = np.max(marks)
minimum_marks = np.min(marks)

print("Maximum Marks:", maximum_marks)
print("Minimum Marks:", minimum_marks)

Maximum Marks: 95
Minimum Marks: 67


In [11]:
std_deviation = np.std(marks)

print("Standard Deviation:", round(std_deviation, 2))

Standard Deviation: 8.57


In [12]:
total_marks = np.sum(marks)

print("Total Marks:", total_marks)

Total Marks: 825


# Results

The student marks dataset was successfully analyzed using NumPy.

| Statistic | Result |
|---|---:|
| Total Marks | 825 |
| Mean Marks | 82.5 |
| Median Marks | 83.0 |
| Maximum Marks | 95 |
| Minimum Marks | 67 |
| Standard Deviation | 8.50 |

# Explanation

- **Mean:** The average score of all students is 82.5.
- **Median:** The middle value of the sorted marks is 83.
- **Maximum:** The highest score is 95.
- **Minimum:** The lowest score is 67.
- **Standard Deviation:** The scores typically vary around the mean, with a population standard deviation of approximately 8.5.

# Conclusion

NumPy simplifies numerical calculations and statistical analysis. This task demonstrates how NumPy arrays and built-in functions can be used to calculate important statistics from student marks.